### DISCUSSION TITLE
trouble with python version selection

### DISCUSSION THREAD

**Discussion by mshqn** [quarto-dev/quarto-cli#4616] [Q&A] [answered]
I am new to Quarto and I have a problem with python versions. Will be very grateful for help! I was trying to render example.qmd and I had an error ``ModuleNotFoundError: No module named 'nbclient'`` which I certainly have installed. So I checked ``quarto check jupyter``:

```
[>] Checking Python 3 installation....OK
      Version: 3.10.10
      Path: C:/Users/iiaba/AppData/Local/Programs/Python/Python310/python.exe
      Jupyter: 5.2.0
      Kernels: conda-env-anaconda3-py, conda-env-pioneer_env2-py, conda-env-pioneer_pr-py, conda-root-py, python3, ir

(-) Checking Jupyter engine render....Traceback (most recent call last):
  File "C:\Users\iiaba\AppData\Local\Programs\Quarto\share\jupyter\jupyter.py", line 21, in <module>
    from notebook import notebook_execute, RestartKernel
  File "C:\Users\iiaba\AppData\Local\Programs\Quarto\share\jupyter\notebook.py", line 18, in <module>
    from nbclient import NotebookClient
ModuleNotFoundError: No module named 'nbclient'
[>] Checking Jupyter engine render....OK
```
But I have already changed env var ``QUARTO_PYTHON``:
![image](https://user-images.githubusercontent.com/58361053/222746710-4b9af07a-b627-42a2-9a43-908617faeb23.png)

And I don't understand why in the Path above I still have this Python310 file. Could someone please recommend a way to fix a Python version for Quarto?

**Comment by mcanouil** [comment]
Could you provide more information on your configuration, such as OS, IDE, etc.?

**Reply by mshqn** [reply]
I am using Windows 10 Pro, VS Code 1.76.0, Quarto 1.2.335

**Comment by mcanouil** [comment]
Could you run in the same context/environment/path, the following two commands and give the exact output?

```powershell
$env:QUARTO_PYTHON
quarto check jupyter
```

**Comment by mshqn** [comment, marked as answer]
The problem was that in one of my Path directories I had another Python executable. So you need to check carefully everything you have in Path, I didn't do that. Hope this will be useful for someone.
